daemon/cluster/cluster.go

```diff
@@ -913,6 +913,12 @@ func (c *Cluster) UpdateService(serviceID string, version uint64, spec types.Ser
 		return err
 	}
 
+	// serviceID could be either ID or name so lookup is needed
+	currentService, err := getService(ctx, c.client, serviceID)
```

Comment:
Thanks @tonistiigi. The pull request has been updated. Please let me know if there are any other issues.